napari/_qt/widgets/qt_dims.py

```diff
@@ -130,20 +131,35 @@ def _update_nsliders(self):
 
     def _resize_axis_labels(self):
         """When any of the labels get updated, this method updates all label
-        widths to the width of the longest label. This keeps the sliders
-        left-aligned and allows the full label to be visible at all times,
-        with minimal space, without setting stretch on the layout.
+        widths to a minimum size. This allows the full label to be
+        visible at all times, with minimal space, without setting stretch on
+        the layout.
         """
-        fm = QFontMetrics(QFont("", 0))
-        labels = self.findChildren(QLineEdit, 'axis_label')
-        newwidth = max(fm.boundingRect(lab.text()).width() for lab in labels)
-
-        if any(self._displayed_sliders):
+        displayed_sliders_idx = [
+            idx
+            for idx, displayed in enumerate(self._displayed_sliders)
+            if displayed
+        ]
+        if displayed_sliders_idx:
+            fm = QFontMetrics(QFont("", 0))
+            labels = self.findChildren(QLineEdit, 'axis_label')
+            displayed_labels = [
+                self.slider_widgets[slider_idx].axis_label
+                for slider_idx in displayed_sliders_idx
+            ]
```

Comment:
Minor: the list of indices is only used once and accessing the widgets should be cheap, so just combine the statements to make this more concise.

```suggestion
        displayed_labels = [
            self.slider_widgets[idx].axis_label
            for idx, displayed in enumerate(self._displayed_sliders)
            if displayed
        ]
        if displayed_labels:
            fm = QFontMetrics(QFont("", 0))
            labels = self.findChildren(QLineEdit, 'axis_label')
```